mediapipe/tasks/ios/vision/core/utils/sources/MPPImage+Utils.mm

```diff
@@ -214,29 +210,91 @@ @implementation MPPCGImageUtils
     if (srcData) {
       // We have drawn the image as an RGBA image with 8 bitsPerComponent and hence can safely input
       // a pixel format of type kCVPixelFormatType_32RGBA for conversion by vImage.
-      pixelDataToReturn = [MPPPixelDataUtils rgbPixelDataFromPixelData:srcData
-                                                             withWidth:width
-                                                                height:height
-                                                                stride:bytesPerRow
-                                                     pixelBufferFormat:kCVPixelFormatType_32RGBA
-                                                                 error:error];
+      imageFrame = [MPPPixelDataUtils imageFrameFromPixelData:srcData
+                                                    withWidth:width
+                                                       height:height
+                                                       stride:bytesPerRow
+                                            pixelBufferFormat:kCVPixelFormatType_32RGBA
+                                                        error:error];
     }
 
     CGContextRelease(context);
   }
 
   CGColorSpaceRelease(colorSpace);
 
-  if (!pixelDataToReturn) {
-    return nullptr;
+  return imageFrame;
+}
+
++ (CGImageRef)cgImageFromImageFrame:(std::shared_ptr<ImageFrame>)imageFrame
+                shouldCopyPixelData:(BOOL)shouldCopyPixelData
+                              error:(NSError **)error {
+  CGBitmapInfo bitmapInfo = kCGImageAlphaNoneSkipLast | kCGBitmapByteOrderDefault;
+
+  ImageFrame *internalImageFrame = imageFrame.get();
+  size_t channelCount = 4;
+
+  switch (internalImageFrame->Format()) {
+    case ImageFormat::SRGBA: {
+      bitmapInfo = kCGImageAlphaPremultipliedLast | kCGBitmapByteOrder32Big;
+      break;
+    }
+    default:
+      [MPPCommonUtils createCustomError:error
+                               withCode:MPPTasksErrorCodeInternalError
+                            description:@"Unsupported Image Format Conversion."];
```

Comment:
Can you add the provided format name?

Reply:
I have changed this to: "Some internal error occurred." I think this is more apt since the function converts an output image frame to an iOS image for tasks like the face styliser. Most likely this won't happen, unless there is some issue in the library source code.